aws_db_parameter_group: Can't modify AWS DB Parameter Group

I can not modifying 'aws_db_parameter_group'.

### Step to Reproduce
1 Apply first tf

```
provider "aws" {
    region = "us-east-1"
}

resource "aws_db_parameter_group" "default" {
    name = "rds-pg-test1"
    family = "mysql5.6"
    description = "test parameter group"

    parameter {
        name = "character_set_server"
        value = "utf8"
    }

    parameter {
        name = "character_set_client"
        value = "utf8"
    }

    parameter {
        name = "collation_server"
        value = "utf8_unicode_ci"
    }

    parameter {
        name = "collation_connection"
        value = "utf8_unicode_ci"
    }
}

resource "aws_db_instance" "default" {
    identifier = "db-test1"
    allocated_storage = "5"
    engine = "mysql"
    engine_version = "5.6.22"
    instance_class = "db.t2.micro"
    storage_type = "gp2"

    username = "root"
    password = "123456789abcdefg"

    parameter_group_name = "${aws_db_parameter_group.default.id}"
}
```

2 Apply modified tf

```
provider "aws" {
    region = "us-east-1"
}

resource "aws_db_parameter_group" "default" {
    name = "rds-pg-test1"
    family = "mysql5.6"
    description = "test parameter group"

    parameter {
        name = "character_set_server"
        value = "utf8mb4"
    }

    parameter {
        name = "character_set_client"
        value = "utf8"
    }

    parameter {
        name = "collation_server"
        value = "utf8_unicode_ci"
    }

    parameter {
        name = "collation_connection"
        value = "utf8_unicode_ci"
    }
}

resource "aws_db_instance" "default" {
    identifier = "db-test1"
    allocated_storage = "5"
    engine = "mysql"
    engine_version = "5.6.22"
    instance_class = "db.t2.micro"
    storage_type = "gp2"

    username = "root"
    password = "123456789abcdefg"

    parameter_group_name = "${aws_db_parameter_group.default.id}"
}
```
```
$ terraform apply                                                                                                                                                    
aws_db_parameter_group.default: Refreshing state... (ID: rds-pg-test1)
aws_db_instance.default: Refreshing state... (ID: db-test1)
aws_db_parameter_group.default: Modifying...
  parameter.1706463059.apply_method: "" => ""
  parameter.1706463059.name:         "collation_connection" => "collation_connection"
  parameter.1706463059.value:        "utf8_unicode_ci" => "utf8_unicode_ci"
  parameter.1895344133.apply_method: "" => ""
  parameter.1895344133.name:         "" => "character_set_server"
  parameter.1895344133.value:        "" => "utf8mb4"
  parameter.2421266705.apply_method: "" => ""
  parameter.2421266705.name:         "character_set_server" => ""
  parameter.2421266705.value:        "utf8" => ""
  parameter.2475805061.apply_method: "" => ""
  parameter.2475805061.name:         "collation_server" => "collation_server"
  parameter.2475805061.value:        "utf8_unicode_ci" => "utf8_unicode_ci"
  parameter.2478663599.apply_method: "" => ""
  parameter.2478663599.name:         "character_set_client" => "character_set_client"
  parameter.2478663599.value:        "utf8" => "utf8"
aws_db_parameter_group.default: Error: 1 error(s) occurred:

* Error modifying DB Parameter Group: InvalidParameterValue: Parameter name cannot be empty
        status code: 400, request id: [334729bd-29d6-11e5-a10d-cdbad18114ff]
Error applying plan:

1 error(s) occurred:

* Error modifying DB Parameter Group: InvalidParameterValue: Parameter name cannot be empty
        status code: 400, request id: [334729bd-29d6-11e5-a10d-cdbad18114ff]

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure
```

### Test code

```
diff --git a/builtin/providers/aws/resource_aws_db_parameter_group_test.go b/builtin/providers/aws/resource_aws_db_parameter_group_test.go
index 93e74bb..3ab20e1 100644
--- a/builtin/providers/aws/resource_aws_db_parameter_group_test.go
+++ b/builtin/providers/aws/resource_aws_db_parameter_group_test.go
@@ -77,6 +77,39 @@ func TestAccAWSDBParameterGroup_basic(t *testing.T) {
                                                "aws_db_parameter_group.bar", "parameter.2478663599.value", "utf8"),
                                ),
                        },
+                       resource.TestStep{
+                               Config: testAccAWSDBParameterGroupModifyParametersConfig,
+                               Check: resource.ComposeTestCheckFunc(
+                                       testAccCheckAWSDBParameterGroupExists("aws_db_parameter_group.bar", &v),
+                                       testAccCheckAWSDBParameterGroupAttributes(&v),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "name", "parameter-group-test-terraform"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "family", "mysql5.6"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "description", "Test parameter group for terraform"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.2450940716.name", "collation_connection"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.2450940716.value", "utf8_general_ci"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.1708034931.name", "character_set_results"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.1708034931.value", "utf8"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.2421266705.name", "character_set_server"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.2421266705.value", "utf8"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.1680942586.name", "collation_server"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.1680942586.value", "utf8_general_ci"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.2478663599.name", "character_set_client"),
+                                       resource.TestCheckResourceAttr(
+                                               "aws_db_parameter_group.bar", "parameter.2478663599.value", "utf8"),
+                               ),
+                       },
                },
        })
 }
@@ -241,6 +274,34 @@ resource "aws_db_parameter_group" "bar" {
 }
 `
 
+const testAccAWSDBParameterGroupModifyParametersConfig = `
+resource "aws_db_parameter_group" "bar" {
+       name = "parameter-group-test-terraform"
+       family = "mysql5.6"
+       description = "Test parameter group for terraform"
+       parameter {
+         name = "character_set_server"
+         value = "utf8"
+       }
+       parameter {
+         name = "character_set_client"
+         value = "utf8"
+       }
+       parameter{
+         name = "character_set_results"
+         value = "utf8"
+       }
+       parameter {
+         name = "collation_server"
+         value = "utf8_general_ci"
+       }
+       parameter {
+         name = "collation_connection"
+         value = "utf8_general_ci"
+       }
+}
+`
+
 const testAccAWSDBParameterGroupOnlyConfig = `
 resource "aws_db_parameter_group" "bar" {
        name = "parameter-group-test-terraform"
```

```
$ cd builtin/providers/aws
$ TF_ACC=1 go test -v -run TestAccAWSDBParameterGroup_basic
=== RUN TestAccAWSDBParameterGroup_basic
--- FAIL: TestAccAWSDBParameterGroup_basic (3.79s)
        testing.go:135: Step 2 error: Error applying: 1 error(s) occurred:

                * 1 error(s) occurred:

                * 1 error(s) occurred:

                * Error modifying DB Parameter Group: InvalidParameterValue: Parameter name cannot be empty
                        status code: 400, request id: [fdef5354-29ce-11e5-84c1-93153ad38a4b]
FAIL
exit status 1
FAIL    github.com/hashicorp/terraform/builtin/providers/aws    3.792s
```

--- a/builtin/providers/aws/resource_aws_db_parameter_group.go
+++ b/builtin/providers/aws/resource_aws_db_parameter_group.go
@@ -166,7 +166,7 @@ func resourceAwsDbParameterGroupUpdate(d *schema.ResourceData, meta interface{})
 				Parameters:           parameters,
 			}
 
-			log.Printf("[DEBUG] Modify DB Parameter Group: %#v", modifyOpts)
+			log.Printf("[DEBUG] Modify DB Parameter Group: %s", modifyOpts)
 			_, err = rdsconn.ModifyDBParameterGroup(&modifyOpts)
 			if err != nil {
 				return fmt.Errorf("Error modifying DB Parameter Group: %s", err)

--- a/builtin/providers/aws/structure.go
+++ b/builtin/providers/aws/structure.go
@@ -186,12 +186,16 @@ func expandIPPerms(
 // Takes the result of flatmap.Expand for an array of parameters and
 // returns Parameter API compatible objects
 func expandParameters(configured []interface{}) ([]*rds.Parameter, error) {
-	parameters := make([]*rds.Parameter, 0, len(configured))
+	var parameters []*rds.Parameter
 
 	// Loop over our configured parameters and create
 	// an array of aws-sdk-go compatabile objects
 	for _, pRaw := range configured {
 		data := pRaw.(map[string]interface{})
+
+		if data["name"].(string) == "" {
+			continue
+		}
 
 		p := &rds.Parameter{
 			ApplyMethod:    aws.String(data["apply_method"].(string)),

